src/main/java/bisq/asset/StellarTokenAddressValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset;
+
+import java.net.URL;
+import java.net.URLConnection;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+/**
+ * Validates whether a stellar address has the necessary trustline
+ * established to receive the token. The validator uses the stellar
+ * horizon network to do the lookup.
+ *
+ * @author Peter Molnar/litlp007
+ * @since 0.7.0
+ */
+public class StellarTokenAddressValidator implements AddressValidator {
+
+   private String code;
+   private String issuer;
+
+   public static String getText(String url) throws Exception {
+        URL website = new URL(url);
+        URLConnection connection = website.openConnection();
+        BufferedReader in = new BufferedReader(
```

Comment:
I don't think we want to make any external queries to validate anything. 
Validation must work offline.

Reply:
Thank you for the quick response.
I understand your position, but let me explain the situation a bit more so you may reconsider.

The ripple/stellar network works in a way that there is no way to tell offline whether a specific address can receive a specific token or not (an account needs to set up a "trust line" before it can hold a specific token).
We could validate the address syntax offline but if someone would buy the token without the established trust line then the seller would not be able to transfer the tokens even if he/she wanted to and the deal would go straight to arbitration (as the seller could not contact the buyer directly to tell him/her to establish the needed trust).

Given the above would you reconsider the rejection - or suggest an alternative?

Thanks!

Reply:
Thanks for the explanation. I still would prefer than the offline online validation and risk some arbitration cases rather to can clear-net URL connections. All network connection in Bisq run over Tor to avoid leaking the IP via clear-net calls.

Reply:
Thank you, I got your point now. I will make a new pull request with offline validation.